src/vcpkg/registries.cpp

```diff
@@ -276,24 +276,23 @@ namespace
                                                          StringView identifier = {});
 
     void load_all_port_names_from_registry_versions(std::vector<std::string>& out,
-                                                    const VcpkgPaths& paths,
+                                                    const Filesystem& fs,
                                                     const path& port_versions_path)
     {
-        for (auto super_directory : stdfs::directory_iterator(port_versions_path))
+        for (auto super_directory : fs.get_directories_non_recursive(port_versions_path, VCPKG_LINE_INFO))
         {
-            if (!vcpkg::is_directory(paths.get_filesystem().status(VCPKG_LINE_INFO, super_directory))) continue;
-
-            for (auto file : stdfs::directory_iterator(super_directory))
+            for (auto file : fs.get_regular_files_non_recursive(super_directory, VCPKG_LINE_INFO))
             {
-                auto filename = vcpkg::u8string(file.path().filename());
-                if (!Strings::ends_with(filename, ".json")) continue;
+                auto filename = vcpkg::u8string(file.filename());
+                if (!Strings::case_insensitive_ascii_ends_with(filename, ".json")) continue;
```

Comment:
Please review this edit closely.

Reply:
LGTM

Reply:
I actually changed my mind on this: I think we want to require that the extension is exactly ".json" on all platforms, and reject ".JsOn" and ".jsON" etc.

Reply:
Added better reporting.